lightning/src/ln/peer_handler.rs

```diff
@@ -340,25 +373,33 @@ pub type SimpleRefPeerManager<'a, 'b, 'c, 'd, 'e, 'f, 'g, SD, M, T, F, C, L> = P
 /// you're using lightning-net-tokio.
 ///
 /// [`read_event`]: PeerManager::read_event
-pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> where
+pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, UMH: Deref, T> where
 		CM::Target: ChannelMessageHandler,
 		RM::Target: RoutingMessageHandler,
-		L::Target: Logger {
+		L::Target: Logger,
+		UMH::Target: UnknownMessageHandler<T> + CustomMessageReader<T>,
+		T: ::std::fmt::Debug {
```

Comment:
I don't get why we need the `T` here. Even if we keep the generic type for unknown messages, I think it should be an associated type on the `UnknownMessageReader`.